core/conversion/converters/impl/shuffle.cpp

```diff
@@ -67,33 +67,33 @@ static auto shuffle_registrations TORCHTRT_UNUSED =
         .pattern(
             {"aten::reshape(Tensor self, int[] shape) -> (Tensor)",
              [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
-               auto in = args[0].ITensorOrFreeze(ctx);
-               auto in_shape = util::toVec(in->getDimensions());
-               std::vector<int64_t> new_shape;
-               if (ctx->input_is_dynamic) {
-                 new_shape = util::toVec(args[1].unwrapToIntList().vec());
-                 int nbDynamicDims = 0;
-                 for (size_t i = 0; i < new_shape.size(); i++) {
-                   if (in_shape[i] == -1)
-                     nbDynamicDims++;
-                 }
-                 if (nbDynamicDims > 1) {
-                   TORCHTRT_THROW_ERROR(
-                       "Resize is currently not supported when target shape contains more than one dynamic dimension");
-                 }
-               } else {
-                 new_shape = torch::reshape(torch::rand(in_shape), args[1].unwrapToIntList().vec()).sizes().vec();
-               }
-
-               auto shuffle = ctx->net->addShuffle(*in);
-               TORCHTRT_CHECK(shuffle, "Unable to create shuffle layer from node: " << *n);
-               shuffle->setReshapeDimensions(util::toDims(new_shape));
-               shuffle->setName(util::node_info(n).c_str());
-
-               auto out_tensor = ctx->AssociateValueAndTensor(n->outputs()[0], shuffle->getOutput(0));
-               LOG_DEBUG("Output tensor shape: " << out_tensor->getDimensions());
-
-               return true;
+                auto in = args[0].ITensorOrFreeze(ctx);
+                auto in_shape = util::toVec(in->getDimensions());
+                std::vector<int64_t> new_shape;
+                nvinfer1::ITensor* shape_tensor;
+                if (ctx->input_is_dynamic) {
```

Comment:
Does this include the case where the non-batch dimension is -1 wildcard? For instance, [ batch_size, -1, spatial_x, spatial_y] in an NCHW format.

Reply:
Yes, it is handled. Eg: The following works
```
class ViewTest(torch.nn.Module):
    def __init__(self):
        super(ViewTest, self).__init__()

    def forward(self, x):
        return x.view(x.size(0), -1, 4)

model = ViewTest().eval().cuda()

with torch_tensorrt.logging.debug():
    trt_mod = torch_tensorrt.compile(model, inputs=[torch_tensorrt.Input(min_shape=[1, 16, 16],
                                                                         opt_shape=[16, 16, 16],
                                                                         max_shape=[32, 16, 16], dtype=torch.float32)], truncate_long_and_double=True)
```